File: SDKs/twilio.com/twilio_pricing_v2/1.36.0/typescript/src/sdk/models/operations/listvoicecountry.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";
export const LISTVOICECOUNTRY_SERVERS = [
	"https://pricing.twilio.com",
];



export class ListVoiceCountryQueryParams extends SpeakeasyBase {
  @Metadata({ data: "queryParam, style=form;explode=true;name=PageSize" })
  pageSize?: number;
}


export class ListVoiceCountrySecurity extends SpeakeasyBase {
  @Metadata({ data: "security, scheme=true;type=http;subtype=basic" })
  accountSidAuthToken: shared.SchemeAccountSidAuthToken;
}


export class ListVoiceCountryRequest extends SpeakeasyBase {
  @Metadata()
  serverUrl?: string;

  @Metadata()
  queryParams: ListVoiceCountryQueryParams;

  @Metadata()
  security: ListVoiceCountrySecurity;
}


export class ListVoiceCountryListVoiceCountryResponseMeta extends SpeakeasyBase {
  @Metadata({ data: "json, name=first_page_url" })
  firstPageUrl?: string;

  @Metadata({ data: "json, name=key" })
  key?: string;

  @Metadata({ data: "json, name=next_page_url" })
  nextPageUrl?: string;

  @Metadata({ data: "json, name=page" })
  page?: number;

  @Metadata({ data: "json, name=page_size" })
  pageSize?: number;

  @Metadata({ data: "json, name=previous_page_url" })
  previousPageUrl?: string;

  @Metadata({ data: "json, name=url" })
  url?: string;
}


export class ListVoiceCountryListVoiceCountryResponse extends SpeakeasyBase {
  @Metadata({ data: "json, name=countries", elemType: shared.PricingV2VoiceVoiceCountry })
  countries?: shared.PricingV2VoiceVoiceCountry[];

  @Metadata({ data: "json, name=meta" })
  meta?: ListVoiceCountryListVoiceCountryResponseMeta;
}


export class ListVoiceCountryResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  listVoiceCountryResponse?: ListVoiceCountryListVoiceCountryResponse;

  @Metadata()
  statusCode: number;
}

File: SDKs/twilio.com/twilio_pricing_v1/1.37.3/python/src/sdk/models/operations/listvoicecountry.py
from dataclasses import dataclass, field
from typing import List,Optional
from dataclasses_json import dataclass_json
from sdk import utils
from sdk.models import shared


LIST_VOICE_COUNTRY_SERVERS = [
	"https://pricing.twilio.com",
]


@dataclass
class ListVoiceCountryQueryParams:
    page_size: Optional[int] = field(default=None, metadata={'query_param': { 'field_name': 'PageSize', 'style': 'form', 'explode': True }})
    

@dataclass
class ListVoiceCountrySecurity:
    account_sid_auth_token: shared.SchemeAccountSidAuthToken = field(metadata={'security': { 'scheme': True, 'type': 'http', 'sub_type': 'basic' }})
    

@dataclass_json
@dataclass
class ListVoiceCountryListVoiceCountryResponseMeta:
    first_page_url: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('first_page_url') }})
    key: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('key') }})
    next_page_url: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('next_page_url') }})
    page: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('page') }})
    page_size: Optional[int] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('page_size') }})
    previous_page_url: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('previous_page_url') }})
    url: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('url') }})
    

@dataclass_json
@dataclass
class ListVoiceCountryListVoiceCountryResponse:
    countries: Optional[List[shared.PricingV1VoiceVoiceCountry]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('countries') }})
    meta: Optional[ListVoiceCountryListVoiceCountryResponseMeta] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('meta') }})
    

@dataclass
class ListVoiceCountryRequest:
    query_params: ListVoiceCountryQueryParams = field()
    security: ListVoiceCountrySecurity = field()
    server_url: Optional[str] = field(default=None)
    

@dataclass
class ListVoiceCountryResponse:
    content_type: str = field()
    status_code: int = field()
    list_voice_country_response: Optional[ListVoiceCountryListVoiceCountryResponse] = field(default=None)
    
